File: tests/fixtures/repomap/sample.lua
local json = require("json")
local utils = require("utils")

function start_server(port)
    print("Starting on " .. port)
end

function get_user(id)
    return { id = id, name = "User" }
end

local function internal_helper()
    -- private helper
    return true
end

function UserService(db)
    local self = {}
    self.db = db

    function self.find(id)
        return db:query(id)
    end

    return self
end

File: tests/fixtures/repomap/sample.zig
const std = @import("std");
const mem = @import("std").mem;

pub fn startServer(port: u16) !void {
    std.debug.print("Starting on {}\n", .{port});
}

pub const UserService = struct {
    db: *Database,

    pub fn getUser(self: *UserService, id: u32) ?User {
        return self.db.find(id);
    }

    fn internalValidate(self: *UserService, id: u32) bool {
        return id > 0;
    }
};

const Config = struct {
    port: u16,
    host: []const u8,
};

pub fn createApp(config: Config) UserService {
    return UserService{ .db = undefined };
}
